flepimop/gempyor_pkg/src/gempyor/batch/_helpers.py

```diff
@@ -0,0 +1,85 @@
+__all__ = ()
+
+
+from collections.abc import Iterable
+from datetime import datetime, timezone
+from typing import Literal
+
+from ..constants import _JOB_NAME_REGEX
+
+
+def _format_resource_bounds(bounds: dict[Literal["cpu", "memory", "time"], float]) -> str:
+    """
+    Format resource bounds for logging.
+
+    Args:
+        bounds: The resource bounds to format.
+
+    Returns:
+        The formatted resource bounds.
+    """
+    fmts = {
+        "cpu": "{:.2f}",
+        "memory": "{:.2f}MB",
+        "time": "{:.2f}s",
+    }
+    return ", ".join(fmts[k].format(v) for k, v in bounds.items())
+
+
+def _job_name(name: str | None, timestamp: datetime | None) -> str:
+    """
+    Generate a unique human readable job name.
+    Args:
+        name: The config name used as a prefix or `None` for no prefix.
+        timestamp: The timestamp used to make the job name unique or `None` to use the
+            current UTC timestamp.
+    Returns:
+        A job name that is unique and intended for use when submitting to slurm.
+    Raises:
+        ValueError: If `name` does not start with a letter and contains characters other
+            than the alphabet, numbers, underscores or dashes.
+    Examples:
+        >>> from gempyor.batch import _job_name
+        >>> _job_name(None, None)
+        '20241105T153818'
+        >>> _job_name("foobar", None)
+        'foobar-20241105T153831'
+        >>> from datetime import datetime, timezone
+        >>> _job_name(None, datetime(2024, 1, 1, tzinfo=timezone.utc))
+        '20240101T000000'
+    """
+    timestamp = datetime.now(timezone.utc) if timestamp is None else timestamp
+    timestamp = timestamp.strftime("%Y%m%dT%H%M%S")
+    if name is not None and not _JOB_NAME_REGEX.match(name):
+        raise ValueError(f"The given `name`, '{name}', is not a valid safe name.")
+    return f"{name}-{timestamp}" if name else timestamp
```

Comment:
how does this related to file_paths.run_id?

Reply:
`file_paths.run_id` just formats a timestamp, do you want this method to call out to that function for that purpose only?